test/integration/magnetometer_system.cc

```diff
@@ -141,3 +141,146 @@ TEST_F(MagnetometerTest, GZ_UTILS_TEST_DISABLED_ON_WIN32(RotatedMagnetometer))
+  {
+    std::lock_guard<std::mutex> lock(mutex);
+    magnetometerMsgs.push_back(_msg);
+  }
+
+  msgs::Magnetometer at(const size_t index) const
+  {
+    std::lock_guard<std::mutex> lock(mutex);
+    return magnetometerMsgs.at(index);
+  }
+
+  size_t size() const
+  {
+    std::lock_guard<std::mutex> lock(mutex);
+    return magnetometerMsgs.size();
+  }
+
+private:
+  mutable std::mutex mutex;
+  std::vector<msgs::Magnetometer> magnetometerMsgs;
+};
+
+MagnetometerMessageBuffer gaussMessages;
+MagnetometerMessageBuffer teslaMessages;
+
+/////////////////////////////////////////////////
+void gaussMagnetometerCb(const msgs::Magnetometer &_msg)
+{
+  gaussMessages.add(_msg);
+}
+
+/////////////////////////////////////////////////
+void teslaMagnetometerCb(const msgs::Magnetometer &_msg)
+{
+  teslaMessages.add(_msg);
+}
+
+TEST_F(MagnetometerTest,
+  GZ_UTILS_TEST_DISABLED_ON_WIN32(GaussToTeslaCorrection))
+{
+  // First, run the world without the fix enabled and record the magnetic field.
+  ServerConfig serverConfig;
+  const auto sdfFile = std::string(PROJECT_SOURCE_PATH) +
+    "/test/worlds/magnetometer_with_tesla.sdf";
+  serverConfig.SetSdfFile(sdfFile);
+
+  Server gaussServer(serverConfig);
+  EXPECT_FALSE(gaussServer.Running());
+  EXPECT_FALSE(*gaussServer.Running(0));
+
+  const auto topic = "world/magnetometer_sensor/model/magnetometer_model/"
+      "link/link/sensor/magnetometer_sensor/magnetometer";
+
+  // subscribe to magnetometer topic
+  std::unique_ptr<transport::Node> gaussNode =
+    std::make_unique<transport::Node>();
+  gaussNode->Subscribe(topic, &gaussMagnetometerCb);
+
+  // step world and verify magnetometer's detected field
+  // Run the gauss server
+  const size_t iters = 100u;
+  gaussServer.Run(true, iters, false);
+
+  // Now, we need to create a new server where the magnetometer
+  // is correctly publishing using tesla.
+  // Flip the use_tesla_for_magnetic_field field to true
+  sdf::SDFPtr teslaWorldSdf(new sdf::SDF());
+  sdf::init(teslaWorldSdf);
+  ASSERT_TRUE(sdf::readFile(sdfFile, teslaWorldSdf));
+  sdf::ElementPtr root = teslaWorldSdf->Root();
+  EXPECT_TRUE(root->HasElement("world"));
+
+  sdf::ElementPtr world = root->GetElement("world");
+  const std::string pluginName = "gz::sim::systems::Magnetometer";
+  const std::string useTeslaElementName = "use_tesla_for_magnetic_field";
+  bool elementSet = false;
+  for (sdf::ElementPtr plugin = world->GetElement("plugin");
+       plugin;
+       plugin = plugin->GetNextElement("plugin")) {
+        if (plugin->Get<std::string>("name") == pluginName) {
+            // Found the magnetometer plugin, now force the field to true
+            if (plugin->HasElement(useTeslaElementName)) {
+                sdf::ElementPtr teslaElement =
+                  plugin->GetElement(useTeslaElementName);
+                teslaElement->Set(true);
+                elementSet = true;
+                break;
+            }
+        }
+    }
```

Comment:
indentation looks wierd

Reply:
Is there an automatic code formatting script I am missing? I ran `make codecheck`, but do you all have pre-commit hook-esque scripts that do the formatting automatically?